### DISCUSSION TITLE
Windows 10 Framework Extension (VSIX) installation failure

### DISCUSSION THREAD

**Discussion by Arkaine80** [MonoGame/MonoGame#8442] [General]
I had to reload my Windows 10 machine from scratch.
I am using VS2022 and followed the installation guide found here (VSIX extension install). https://docs.monogame.net/articles/getting_started/1_setting_up_your_development_environment_windows.html 

The templates work, but the project doesn't compile. The MonoGame dependencies did not exist and dotnet tool restore failed, so I used the developer command prompt and manually added the Nuget packages that it complained about (dotnet-mcgb*).

It now recognises the Microsoft.XNA.* namespaces and MCGB editor opens up, but when I try run the application (base application, no custom code), I get compile errors "Unable to find the package SharpDX" (also SharpDX.DXGI, SharpDX.Direct2D1, SharpDX.Direct3D11, SharpDX.MediaFoundation, SharpDX.XAudio2, SharpDX.XInout). 

Clearly the VSIX installation did not complete. How can I manually add SharpDX to get MonoGame to work?

I have tried uninstalling and re-installing several times (even with reboots between uninstall and re-install).

**Comment by Arkaine80** [comment]
Same is happening with 3.8.2 release

**Reply by KelsamGames** [reply]
Not sure why VS wouldn't be auto-downloading what you need, but you can manually install the DirectX 9.0c SDK via https://www.microsoft.com/en-us/download/details.aspx?id=6812

**Reply by Arkaine80** [reply]
Thank you so much for replying, however this did not work. Still have the same errors. In case it helps, the full error message is:
**Unable to find package SharpDX. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs, Microsoft Visual Studio Offline Packages**

**Reply by KelsamGames** [reply]
Hmm. My next best shot-in-the-dark guess is that something is messed up with your nuget package manager then. If you right-click on your "Dependencies" and manage your nuget packages, you could try to manually install your missing dependencies. Search for and install each package on the right of this screenshot. Hopefully installing those will pull their own dependencies down as well.

![Screenshot 2024-09-05 105919](https://github.com/user-attachments/assets/bc8b95e1-eabb-434a-b2e5-97a457e3adbc)

**Reply by Arkaine80** [reply]
I tried to do as suggested and found that my nuget package manager source was only offline.
In case anyone else has this issue in future, I did a VS 2022 repair (might not be needed, but should clear Monogame for a fresh re-install), then added in the nuget package store:

1. Tools > NuGet Package Manager > Package Manager Settings;
2. Select Package Sources;
3. Click the + button to add a new package source;
4. Name = nuget.org, Source = https://api.nuget.org/v3/index.json;
5. Click Update or OK to save the changes.
6. Close VS;

Add the MonoGame extension again.
Everything is up and running for DirectX and OpenGL

Thanks for pointing me in the right direction Kelsam - much appreciated!
